### ISSUE TITLE
Error when having CNAME and TXT in the same subdomain

### ISSUE THREAD

**Issue by stavros-k** [octodns/octodns#1143]
Hello, when I do this, I get the below error. currently I work around this with `lenient` 
Is this the only way, or is there something else I'm missing? 
Thanks!

```yaml
subdomain:
- type: CNAME
  value: some.other.dns.here.
  octodns:
    cloudflare:
      proxied: true
- type: TXT
  value: google-site-verification=blabla
```

```python
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.8/x64/bin/octodns-validate", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/octodns/cmds/validate.py", line 44, in main
    manager.validate_configs(lenient=args.all)
  File "/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/octodns/manager.py", line 992, in validate_configs
    source.populate(zone, lenient=lenient)
  File "/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/octodns/provider/yaml.py", line 363, in populate
    self._populate_from_file(source, zone, lenient)
  File "/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/octodns/provider/yaml.py", line 318, in _populate_from_file
    zone.add_record(
  File "/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/octodns/zone.py", line 180, in add_record
    raise InvalidNodeException(
octodns.zone.InvalidNodeException: Invalid state, CNAME at submdomain.domain.com. cannot coexist with other records, ./domains/domain.com.yaml, line 73, column 3
```

Btw this works fine without `lenient`

```yaml
"":
  - type: TXT
    values:
      - v=spf1 include:domain.com -all
      - google-site-verification=blabla
  - type: MX
    values:
    - exchange: mx.domain.com.
      preference: 10
```

**Comment by ross** [comment]
> octodns.zone.InvalidNodeException: Invalid state, CNAME at submdomain.domain.com. cannot coexist with other records, ./domains/domain.com.yaml, line 73, column 3

As the error message you pasted says it's not legal DNS spec-wise for a CNAME to coexist at the same node with other records. Most providers won't accept it and the behavior when they do the behavior may be undefined, e.g. a client who's previously seen the `CNAME` record and still has it cached may not bother querying for another type of record at the same node and instead ask wherever the `CNAME` pointed. 

Cloudflare proxied maybe complicating the picture further or it may be getting rid of the issue, not something I've really dug into as I've not run across proxied `CNAME`s before. I've generally seen them with `A` and/or `AAAA`.  

`lenient` will tell octoDNS not to warn you about it and is probably the best bet if a `CNAME` is what you require at that node.

**Comment by stavros-k** [comment]
I see, I guess I'm too used to dns providers allowing this. Is there a way to define lenient on this subdomain only instead of the whole domain? So it still warns for other similar issues expect this one 

Thanks for taking the time to answer

**Comment by ross** [comment]
> I see, I guess I'm too used to dns providers allowing this. Is there a way to define lenient on this subdomain only instead of the whole domain? So it still warns for other similar issues expect this one

The way you set `lenient` in the original YAML should only apply it to that specific record in just that node. Any other records will still throw errors and anything else below it, e.g. `another.subdomain.whatever` would still throw errors. 

Fwiw even when `lenient` is set it'll still print warnings about the things it fines. I've toyed with the idea of allowing quelling specific validations, but that hasn't ever gotten beyond the thinking about it for 15m stage. 

> Thanks for taking the time to answer

NP.

**Comment by stavros-k** [comment]
Ah there is a record level lenience, I was using the zone level. 
I have been using octodns since lots of versions before so can't remember if it wasn't on the docs or I just missed it. (Probably the latter).

That should do it! Thanks for your time. Feel free to close this.

**Comment by GrantBirki** [comment]
I just ran into this problem as well and so glad that I stumbled across this issue. I'm using `octodns` with cloudflare and the solution listed here worked perfectly for me.

P.S. @ross we have never met but I interact with stuff you have written all the time at work 😉. Just realized that `octodns` was open source and now I'm wiring it up to my personal repo and using [github/branch-deploy](https://github.com/github/branch-deploy) to facilitate the deployment process just as one might do with friendly robot like hubot 🎉
